dbms/src/TiDB/Schema/SchemaBuilder.cpp

```diff
@@ -410,70 +460,68 @@ void SchemaBuilder<Getter, NameMapper>::applyPartitionDiff(
     const TableInfoPtr & table_info,
     const ManageableStoragePtr & storage)
 {
-    const auto & orig_table_info = storage->getTableInfo();
-    if (!orig_table_info.isLogicalPartitionTable())
+    const auto & local_table_info = storage->getTableInfo();
+    // ALTER TABLE t PARTITION BY ... may turn a non-partition table into partition table
+    if (!local_table_info.isLogicalPartitionTable())
     {
-        LOG_ERROR(
+        LOG_INFO(
             log,
-            "old table in TiFlash not partition table {} with database_id={}, table_id={}",
-            name_mapper.debugCanonicalName(*db_info, orig_table_info),
+            "Altering non-partition table to be a partition table {} with database_id={}, table_id={}",
+            name_mapper.debugCanonicalName(*db_info, local_table_info),
             db_info->id,
-            orig_table_info.id);
-        return;
+            local_table_info.id);
     }
```

Comment:
When fetching the last table schema from TiKV, TiFlash may see a non-partition table turned into a table with "partition" field like this. 
We need to create the physical table with the id in `partition.adding_definitions`/`partition.definitions`/`partition.dropping_definitions`
```
"partition": {
        "type": 0,
        "expr": "",
        "columns": null,
        "enable": true,
        "definitions": [{
            "id": 164,
            "name": {
                "O": "pFullTable",
                "L": "pfulltable"
            },
            "less_than": null,
            "in_values": null,
            "policy_ref_info": null,
            "comment": "Intermediate partition during ALTER TABLE ... PARTITION BY ..."
        }],
        "adding_definitions": [{
            "id": 168,
            "name": {
                "O": "p0",
                "L": "p0"
            },
            "less_than": null,
            "in_values": null,
            "policy_ref_info": null
        }, {
            "id": 169,
            "name": {
                "O": "p1",
                "L": "p1"
            },
            "less_than": null,
            "in_values": null,
            "policy_ref_info": null
        }, {
            "id": 170,
            "name": {
                "O": "p2",
                "L": "p2"
            },
            "less_than": null,
            "in_values": null,
            "policy_ref_info": null
        }],
        "dropping_definitions": [{
            "id": 164,
            "name": {
                "O": "pFullTable",
                "L": "pfulltable"
            },
            "less_than": null,
            "in_values": null,
            "policy_ref_info": null,
            "comment": "Intermediate partition during ALTER TABLE ... PARTITION BY ..."
        }],
        "NewPartitionIDs": null,
        "states": null,
        "num": 1,
        "ddl_state": 3,
        "new_table_id": 171,
        "ddl_type": 2,
        "ddl_expr": "`a`",
        "ddl_columns": null
    },
```